src/features/BroadcastTransaction/components/BroadcastTx.tsx

```diff
@@ -132,34 +135,39 @@ const BroadcastTx = ({ signedTx, network, onComplete, handleNetworkChanged }: Pr
           height={'250px'}
           placeholder="0xf86b0284ee6b2800825208944bbeeb066ed09b7aed07bf39eee0460dfa26152088016345785d8a00008029a03ba7a0cc6d1756cd771f2119cf688b6d4dc9d37096089f0331fe0de0d1cc1254a02f7bcd19854c8d46f8de09e457aec25b127ab4328e1c0d24bfbff8702ee1f474"
           onChange={handleChange}
-          onBlur={validateField}
-          inputError={!transaction ? inputError : ''}
+          inputError={userInput.length > 0 ? inputError : ''}
         />
-        {transaction && <IdenticonIcon address={userInput} />}
       </InputWrapper>
-      {transaction ? (
+      {isValid && transaction && (
         <React.Fragment>
           {!transaction.chainId && (
             <NetworkSelectWrapper>
-              <NetworkSelector network={network} onChange={handleNetworkChanged} />
-              {!network && (
+              <NetworkSelector network={networkId} onChange={handleNetworkChanged} />
```

Comment:
Can you change this elsewhere to just be `networkId` so we don't confuse ourselves in the future?